storage/Devices/MdImpl.cc

```diff
@@ -794,7 +817,11 @@ namespace storage
     void
     Md::Impl::do_extend(const BlkDevice* blk_device) const
     {
-	string cmd_line = MDADMBIN " --add " + quote(get_name()) + " " + quote(blk_device->get_name());
+	const MdUser* md_user = blk_device->get_impl().get_single_out_holder_of_type<const MdUser>();
```

Comment:
Should not be nessesary since it is an internal function. If ```blk_device == nullptr``` the actiongraph is severly broken.